### DISCUSSION TITLE
Just ignore fields in request when model is marked strict and do not throw error.

### DISCUSSION THREAD

**Discussion by mevivek** [loopbackio/loopback-next#7206] [Q&A] [answered]
I have created a User model with id. I am not allowing additional properties because I don't want to save extra data in the database. In lb4 `@post` request I am excluding `id` so when I am calling this endpoint from the android application which is sending `User` data class object containing `id` (which is blank), the API is throwing an error. In kotlin, creating another class with all the fields excluding `id` is not feasible. 
Similarly, in post request, the android application is sending the `id` in the body. Even though I have used `Omit<User, 'id'>` utility class on the request object it, is not omitting the `id` field and lb4 is creating a blank id field in the database (i have set forceId: true).

**Comment by marioestradarosa** [comment, marked as answer]
@mevivek , you should remove the id before calling the `create` repository method. 

_Assuming you have the id as follows:_
```ts
export class User extends Entity {
  @property({
    type: 'number',
    id: true,
    generated: true,
  })
  id?: number;
```

Then your controller should look like this:

```ts
async create(
    @requestBody({
      content: {
        'application/json': {
          schema: getModelSchemaRef(User, {
            title: 'NewUser',
          }),
        },
      },
    })
    user: User,
  ): Promise<User> {
    delete user.id;
    return this.userRepository.create(user);
  }
```

Note that I removed the `exclude: ['id'],`. from the `@requestBody` decorator and the `Omit<User, 'id'>` was replaced by `User` in the parameter type.

**Reply by marioestradarosa** [reply]
Also note that on my test, I used the following SQL table creation:

```sql
CREATE TABLE user 
( id  INT(11) NOT NULL AUTO_INCREMENT,
  last_name VARCHAR(30) NOT NULL,
  first_name VARCHAR(25),
  CONSTRAINT contacts_pk PRIMARY KEY (id)
);
```

My user Model looks like this:

```ts
@model({name: 'user'})
export class User extends Entity {
  @property({
    type: 'number',
    id: true,
    generated: true,
  })
  id?: number;

  @property({
    type: 'string',
    required: true,
    mysql: {column: 'first_name'}
  })
  firstName: string;

  @property({
    type: 'string',
    required: true,
    mysql: {column: 'last_name'}
  })
  lastName: string;


```

**Reply by mevivek** [reply]
Thanks! @marioestradarosa. Your solution worked. But it will disturb OpenAPI docs. And why Loopback doesn't handle this scenario. If I am 'excluding some field from the model schema then loopback should delete them from the request body and should not throw an error.

**Reply by marioestradarosa** [reply]
@mevivek ,  here is how it works.  

The following code is allowing the generation of appropriate OpenApi documentation, saying it will not receive a property name id. The ```requestBody``` is just a decorator for this metadata, but at execution time, it is the responsability of the ```request-body.validator.ts``` to make sure this is enforced.

```ts
 async create(
    @requestBody({
      content: {
        'application/json': {
          schema: getModelSchemaRef(Todo, {
            title: 'NewTodo',
            exclude: ['id'],
          }),
        },
      },
    })
```
In the same line, you find the following statement. the ```Omit``` is not part of loopback 4 framework, it is a type script `TYPE` generator, in other words it is just creating a type of the same shape as `Todo` but  without the id property, and is working fine.  

```ts
 todo: Omit<Todo, "id">,
  ): Promise<Todo> {
     return this.todoRepository.create(todo);
  }
  ```
  
The latter `TRUST` the first one, so it is just assigning whatever is coming from the request to the new TYPE without the id, but this assignment overrides the whole object, why ??, well, remember that type script types are not available at runTime. 
  
  So, both steps must agree and they work fine. However, in your case, you can't go with this Open API spec because your client code is always sending the id, so in order for your client code to conform to the Open API spec of any REST api, the server Open API should not say exclude id, because you will always be sending it, and that will move you to the next step, otherwise you will never pass over the first one, which is enforced by the request body validator. 
  
  After you pass the first step, then the only way I found was to remove the id property manually, since Omit<> is just creating a new type not removing or assigning anything. I believe with this, you conform to the Open API spec _(according to my initial example)_  and make a little tweak in order to remove the id, making it transparently for the client application. 
  
  Also you can use the omit method from `lodash`, so your code can look like so:
  
  ```ts
import * as _ from 'lodash';

and then ....
       return this.todoRepository.create(_.omit(todo,['id']));
  
  ```
